src/AutoRest.CSharp/DataPlane/AutoRest/DataPlaneTarget.cs

```diff
@@ -25,6 +21,10 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
             var headerModelModelWriter = new DataPlaneResponseHeaderGroupWriter();
             var longRunningOperationWriter = new LongRunningOperationWriter();
 
+            var helperWriter = new CodeWriter();
+            new ExpressionTypeProviderWriter(helperWriter, OptionalTypeProvider.Instance).Write();
+            project.AddGeneratedFile($"Internal/{OptionalTypeProvider.Instance.Type.Name}.cs", helperWriter.ToString());
```

Comment:
I suggest we could add a method such as `AddHelperFile` on `GeneratedCodeWorkspace` so that we could do something different from usual generated files, for instance in there we could let the post-processor to ignore all helper files.

Reply:
Having this in here lets the post processor omit any helpers that aren't actually used.